src/application/services/useNoteSettings.ts

```diff
@@ -65,14 +65,18 @@ export default function (): UseNoteSettingsComposableState {
   };
 
   /**
-   * Update note settings
+   * Update field isPublic in note settings
    *
    * @param id - Note id
-   * @param data - Note settings data with new values
+   * @param newIsPublicValue - new isPublic
    */
-  const update = async (id: NoteId, data: Partial<NoteSettings>): Promise<void> => {
-    noteSettings.value = await noteSettingsService.patchNoteSettingsByNoteId(id, data);
-  };
+  async function updateIsPublic(id: NoteId, newIsPublicValue: boolean): Promise<void> {
+    const response = await noteSettingsService.patchNoteSettingsByNoteId(id, { isPublic: newIsPublicValue });
```

Comment:
`response` is a transport-related term, it should not be used in this context.

Reply:
You mean I should rename it?

Reply:
service returns the note settings, you can pick the new value using desctructor

```ts
const { isPublic } = await ...
```